src/runtime/windows_tempfile.cpp

```diff
@@ -0,0 +1,43 @@
+#include "HalideRuntime.h"
+
+#ifdef BITS_64
+#define WIN32API
+#else
+#define WIN32API __stdcall
+#endif
+
+#define MAX_PATH 260
+
+extern "C" {
+
+extern uint32_t WIN32API GetTempPathA(uint32_t, char *);
+extern uint32_t WIN32API GetTempFileNameA(const char *, const char *, uint32_t, char *);
+
+WEAK int halide_create_temp_file(void *user_context, const char *prefix, const char *suffix,
+                                 char *path_buf, size_t path_buf_size) {
+    // Windows implementations of mkstemp() try to create the file in the root
+    // directory, which is... problematic.
+    char tmp_dir_path[MAX_PATH];
+    uint32_t ret = GetTempPathA(MAX_PATH, tmp_dir_path);
+    if (ret != 0) {
+        return halide_error_code_internal_error;
+    }
+    // GetTempFileName doesn't allow us to specify a custom suffix, so if
+    // on is requested, fail.
+    if (strlen(suffix) > 0) {
```

Comment:
I think if suffix is NULL, this will crash. I vaguely remember this happening with strlen to me a while ago. Is the intent that "" is passed for no suffix? I think it might be best to just change the if to `suffix && strlen(suffix) > 0`.